src/gb/audio.c

```diff
@@ -627,19 +629,22 @@ void GBAudioSamplePSG(struct GBAudio* audio, int16_t* left, int16_t* right) {
 	}
 
 	if (audio->playingCh4 && !audio->forceDisableCh[3]) {
-		int8_t sample = _coalesceNoiseChannel(&audio->ch4);
+		int16_t sample = _coalesceNoiseChannel(&audio->ch4);
 		if (audio->ch4Left) {
-			sampleLeft += sample;
+			sampleLeftUnscaled += sample;
 		}
 
 		if (audio->ch4Right) {
-			sampleRight += sample;
+			sampleRightUnscaled += sample;
 		}
 	}
 
 	sampleLeft <<= 3;
 	sampleRight <<= 3;
 
+	sampleLeft += sampleLeftUnscaled;
```

Comment:
It may make sense to just move the upshift to before adding in channel 4 instead of managing two sets of variables.